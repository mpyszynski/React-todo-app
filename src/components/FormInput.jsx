import React from 'react';
import './style/form.css';
import PropTypes from 'prop-types';

export default function FormInput({
  label,
  value,
  name,
  placeholder,
  onChange,
}) {
  return (
    <section className='form__section'>
      <label>{label}</label>
      <input
        type='text'
        value={value}
        name={name}
        placeholder={placeholder}
        onChange={onChange}
        className='form__input'
        required
        minLength={3}
      />
    </section>
  );
}

FormInput.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
