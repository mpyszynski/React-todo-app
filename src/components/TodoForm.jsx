import React from 'react';
import FormInput from './FormInput';
import './style/form.css';
import PropTypes from 'prop-types';

export default function TodoForm(props) {
  const heading =
    props.listLength < 1 ? (
      <h1>It seems that your to do list is empty. Let's add first item!</h1>
    ) : (
      <h1>Add more items to your to do list.</h1>
    );
  return (
    <form onSubmit={props.handleSubmit} className='form__main'>
      {heading}
      <FormInput
        label='Title'
        value={props.title}
        name='title'
        placeholder='Title...'
        onChange={props.handleChange}
      />
      <FormInput
        label='Description'
        value={props.description}
        name='description'
        placeholder='Description...'
        onChange={props.handleChange}
      />
      <input type='submit' value='Add' className='form__button' />
    </form>
  );
}

TodoForm.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  listLength: PropTypes.number.isRequired,
};
