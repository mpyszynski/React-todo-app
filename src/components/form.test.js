import React from 'react';
import { shallow } from 'enzyme';
import FormInput from './FormInput';
import TodoForm from './TodoForm';

const mockFunction = jest.fn();

const testState = { title: 'Buy milk', description: 'Go to kiwi and buy milk' };

describe('<FormInput />', () => {
  test('should return section with label and input', () => {
    const wrapper = shallow(
      <FormInput
        label='Title'
        value={testState.title}
        name='title'
        placeholder='Title...'
        onChange={e => {
          testState[e.target.name] = e.target.value;
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('label').text()).toEqual('Title');
    expect(wrapper.find('.form__input').props().value).toEqual('Buy milk');
    expect(wrapper.find('.form__input').props().name).toEqual('title');
    expect(wrapper.find('.form__input').props().placeholder).toEqual(
      'Title...',
    );
    wrapper.find('.form__input').simulate('change', {
      target: { name: 'title', value: 'Go for a walk' },
    });
    expect(testState.title).toEqual('Go for a walk');
  });
});

describe('<TodoForm />', () => {
  test('should return correct heading with 0 items in todo list', () => {
    const wrapper = shallow(
      <TodoForm
        title={testState.title}
        description={testState.description}
        handleChange={mockFunction}
        handleSubmit={mockFunction}
        listLength={0}
      />,
    );
    expect(wrapper.find('h1').text()).toEqual(
      "It seems that your to do list is empty. Let's add first item!",
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('should return correct heading with 10 items in todo list', () => {
    const wrapper = shallow(
      <TodoForm
        title={testState.title}
        description={testState.description}
        handleChange={mockFunction}
        handleSubmit={mockFunction}
        listLength={10}
      />,
    );
    expect(wrapper.find('h1').text()).toEqual(
      'Add more items to your to do list.',
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('should have 4 children', () => {
    const wrapper = shallow(
      <TodoForm
        title={testState.title}
        description={testState.description}
        handleChange={mockFunction}
        handleSubmit={mockFunction}
        listLength={10}
      />,
    );
    expect(wrapper.find('.form__main').children().length).toEqual(4);
  });
});
