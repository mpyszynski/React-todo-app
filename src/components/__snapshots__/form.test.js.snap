// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> should return section with label and input 1`] = `ShallowWrapper {}`;

exports[`<TodoForm /> should return correct heading with 0 items in todo list 1`] = `ShallowWrapper {}`;

exports[`<TodoForm /> should return correct heading with 10 items in todo list 1`] = `ShallowWrapper {}`;
